### PR TITLE
Replace PXRUSDMAYA_DEFINE_READER macro with class declaration/definition to make the code easier to read and debug.

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
### Description of problem:
When using PXRUSDMAYA_DEFINE_READER Macro it is a bit hard to follow the code logic and debug.

### CODE REVIEW

**Comment by mattyjams** [comment]
Looks good to me, thanks @HamedSabri-adsk!

For some historical context, `PXRUSDMAYA_DEFINE_READER` was the initial way of declaring a reader before we had a registry for import more like the one we have for export. Ideally, everything still using `PXRUSDMAYA_DEFINE_READER` would get converted to being class-based and we'd get rid of the `PXRUSDMAYA_DEFINE_READER` macro, but I'm ok doing that conversion in pieces.

**Comment by HamedSabri-adsk** [comment]
@mattyjams Great and thanks for the historical context.
